src/search/evaluators/g_evaluator.cc

```diff
@@ -22,7 +47,7 @@ static shared_ptr<Evaluator> _parse(OptionParser &parser) {
     if (parser.dry_run())
         return nullptr;
     else
-        return make_shared<GEvaluator>();
+        return make_shared<GEvaluator>(tasks::g_root_task);
```

Comment:
This code is only relevant if users use g() on the commandline. It is unrelated to the cost_type option of the search engines. It can still lead to confusion though, I agree.